src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -61,7 +70,7 @@ public static bool IsXmsPageableOrArrayResponseOperation(Operation op, ServiceDe
                     return false;
                 }
 
-                if (entity.Definitions[modelLink.StripDefinitionPath()].Properties?.Values?.Any(type => type.Type == DataType.Array)??false)
+                if (entity.Definitions[modelLink.StripDefinitionPath()].Properties?.Values?.Any(type => type.Type == DataType.Array) ?? false)
```

Comment:
should we refactor code a bit and return the result of evaluating the conditions? return ((entity.Definitions[modelLink.StripDefinitionPath()].Properties?.Values?.Count ?? 2) < 2) && (entity.Definitions[modelLink.StripDefinitionPath()].AllOf == null) && ((entity.Definitions[modelLink.StripDefinitionPath()].Properties?.Values?.Any(type => type.Type == DataType.Array))